routers/api/packages/conda/conda.go

```diff
@@ -0,0 +1,313 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package conda
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	packages_model "code.gitea.io/gitea/models/packages"
+	conda_model "code.gitea.io/gitea/models/packages/conda"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	packages_module "code.gitea.io/gitea/modules/packages"
+	conda_module "code.gitea.io/gitea/modules/packages/conda"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/routers/api/packages/helper"
+	packages_service "code.gitea.io/gitea/services/packages"
+
+	"github.com/dsnet/compress/bzip2"
+)
+
+func apiError(ctx *context.Context, status int, obj interface{}) {
+	type Error struct {
+		Reason  string `json:"reason"`
+		Message string `json:"message"`
+	}
+
+	helper.LogAndProcessError(ctx, status, obj, func(message string) {
+		resp := ctx.Resp
+		resp.Header().Set("Content-Type", "application/json")
+		resp.WriteHeader(status)
+		if err := json.NewEncoder(resp).Encode(Error{
+			Reason:  http.StatusText(status),
+			Message: message,
+		}); err != nil {
+			log.Error("JSON encode: %v", err)
+		}
+	})
```

Comment:
So why not use ctx.JSON like in routers/api/packages/composer/composer.go:apiError:

```go
func apiError(ctx *context.Context, status int, obj interface{}) {
	helper.LogAndProcessError(ctx, status, obj, func(message string) {
		type Error struct {
			Status  int    `json:"status"`
			Message string `json:"message"`
		}
		ctx.JSON(status, struct {
			Errors []Error `json:"errors"`
		}{
			Errors: []Error{
				{Status: status, Message: message},
			},
		})
	})
}
```

Reply:
Used Pub as template and somehow missed `ctx.JSON`. Fixed now.